### PR TITLE
Added TeacherCon program registration form

### PR DESCRIPTION
**Pull request by aoby**

### CODE REVIEW

File: cookbooks/cdo-varnish/libraries/http_cache.rb [hunk 1]
```diff
@@ -83,6 +83,7 @@ def self.config(env)
               /teacher-dashboard*
               /manage-professional-development-workshops*
               /professional-development-workshop-surveys*
+              /pd-program-registration*
```
Comment (aoby) [inline cookbooks/cdo-varnish/libraries/http_cache.rb:86]: This is needed for `authentication_required!` to work (`index.haml`, line 8 below)

File: pegasus/forms/pd_program_registration.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
+class PdProgramRegistration
+  def self.normalize(data)
+    result = {}
+    result[:user_id_i] = required integer data[:user_id_i]
+    result[:pd_teacher_application_id] = required integer data[:pd_teacher_application_id]
+    result[:first_name_s] = required data[:first_name_s]
```
Comment (mehalshah) [inline pegasus/forms/pd_program_registration.rb]: optional, but you could do
```
[:first_name_s, :last_name_s, ...].each do |string|
result[string] = required data[string]
end
```

File: pegasus/forms/pd_program_registration.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    result[:accept_b] = required data[:accept_b]
+    if result[:accept_b] == '0'
+      result[:accept_no_reason_s] = required data[:accept_no_reason_s]
+      if result[:accept_no_reason_s] == OTHER
+        result[:accept_no_other_s] = data[:accept_no_other_s]
+      end
+    end
+
+    if result[:accept_b] == '1'
+      result[:address_s] = required stripped data[:address_s]
```
Comment (mehalshah) [inline pegasus/forms/pd_program_registration.rb:19]: In fact, you could compress a whole lot of this with similar statements to above. Can talk offline

File: pegasus/sites.v3/code.org/public/js/pd-program-registration.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  $('#error-message').text('An error occurred. Please check to make sure all required fields have been filled out.').show();
+
+  $('body').scrollTop(0);
+  $("#btn-submit").removeAttr('disabled');
+  $("#btn-submit").removeClass("button_disabled").addClass("button_enabled");
+}
+
+function PdProgramRegistrationFormSubmit(event) {
```
Comment (mehalshah) [inline pegasus/sites.v3/code.org/public/js/pd-program-registration.js]: standard is to have this lowercase

Comment (aoby) [reply pegasus/sites.v3/code.org/public/js/pd-program-registration.js]: Renamed to `onPdProgramRegistrationFormSubmit`

**Pull request merged**